cmake/linker/ld/linker_libraries.cmake

```diff
@@ -33,6 +33,8 @@ if(CONFIG_NEWLIB_LIBC AND CMAKE_C_COMPILER_ID STREQUAL "GNU")
   # link dependency to libc (strchr), but libc also has dependencies to libgcc.
   # Lib C depends on libgcc. e.g. libc.a(lib_a-fvwrite.o) references __aeabi_idiv
   set_property(TARGET linker APPEND PROPERTY link_order_library "math;c;rt;c")
-else()
+elseif(NOT CONFIG_MINIMAL_LIBC)
```

Comment:
one intention with new property design is to avoid the linker CMake integration having to know and deal with all possible permutations regarding C libraries and ordering.

Generally only highlevel info, such as `CONFIG_CPP` setting should impact here.

The linker should define the standard for the toolchain, and then if a given CMake C lib integration needs to adjust, then it can do so, like it's done for picolibc.

So this is better handled in the `minimal_libc` CMake code, in this case by clearing the C lib property (because minimal libc is part of `whole-archive`).

As for the newlib case above, then that one is a bit special, as we either end in newlib having to know bit more specifics on the compiler (GNU/non-GNU), or the ld CMake knowing about newlib.

(before you ask, then yes `CONFIG_MINIMAL_LIBCPP` test above should also be moved out of this file)

Moving this to https://github.com/zephyrproject-rtos/zephyr/blob/main/lib/libc/minimal/CMakeLists.txt will also remove the need for checking for `CONFIG_MINIMAL_LIBC` in each `linker_libraries.cmake` file.

Also consider looking at this comment thread: https://github.com/zephyrproject-rtos/zephyr/pull/79629#discussion_r1817235436

Reply:
> before you ask, then yes `CONFIG_MINIMAL_LIBCPP` test above should also be moved out of this file

addressed here" https://github.com/zephyrproject-rtos/zephyr/pull/80621, including handling of `CONFIG_NATIVE_BUILD`.